src/main/drivers/video_max7456.c

```diff
@@ -0,0 +1,620 @@
+
+    // force soft reset on Max7456
+    max7456_write(MAX7456_REG_VM0, MAX7456_VM0_BIT_SOFTWARE_RESET); // without video mode
+    delay(100);
+
+    while(!max7456_isResetComplete()) {
+        resetWait++;
+        delay(10);
+    }
+}
+
+void max7456_hardwareReset(void)
+{
+    gpio_config_t cfg = { MAX7456_NRST_PIN, Mode_Out_PP, Speed_2MHz };
+
+    RCC_AHBPeriphClockCmd(MAX7456_NRST_GPIO_PERIPHERAL, ENABLE);
+
+    gpioInit(MAX7456_NRST_GPIO, &cfg);
+
+    // RESET
+    digitalLo(MAX7456_NRST_GPIO, MAX7456_NRST_PIN);
+    delay(100);
+    digitalHi(MAX7456_NRST_GPIO, MAX7456_NRST_PIN);
+}
+
+void max7456_disableOSD(void)
+{
+    max7456_write(MAX7456_REG_VM0, max7456_videoModeMask); // MAX7456_VM0_BIT_OSD_ENABLE unset.
+}
+
+void max7456_init(videoMode_e videoMode)
+{
+    // device is rated for 10mhz max
+    spiSetDivisor(MAX7456_SPI_INSTANCE, SPI_9MHZ_CLOCK_DIVIDER);
+
+    max7456_setVideoMode(videoMode);
+
+    max7456_softReset();
+
+    max7456_write(MAX7456_REG_OSDM, 0x00);
+    //read black level register
+    uint8_t blackLevelResult = max7456_read(MAX7456_REG_OSDBL_READ);
+
+    // set all rows to same charactor black/white level
+    uint8_t row;
+    for(row = 0; row < max7456_screenRows; row++) {
+        max7456_write(MAX7456_REG_RB0 + row, BWBRIGHTNESS);
+    }
+
+    max7456_write(MAX7456_REG_VM0, MAX7456_VM0_BIT_OSD_ENABLE | max7456_videoModeMask);
+    delay(100);
+
+    // Set black level
+    uint8_t blackLevelValue = (blackLevelResult & 0xef); // Set bit 4 to zero 11101111 (bit is 0 based index)
+
+    max7456_write(MAX7456_REG_OSDBL, blackLevelValue);
+}
+
+#define MAX7456_CHARACTER_BUFFER_SIZE 54
+static void max7456_setFontCharacter(uint8_t characterIndex, const uint8_t *characterBitmap)
+{
+    // cannot update font NVM with OSD enabled.
+    max7456_disableOSD();
+
+    // a short delay is required after disabling the OSD before sending the character index otherwise the CMAH register is not updated.
+    delay(20);
+    max7456_write(MAX7456_REG_CMAH, characterIndex);
+
+    // transfer character bitmap to character buffer in shadow ram
+    for(uint8_t i = 0; i < MAX7456_CHARACTER_BUFFER_SIZE; i++)
+    {
+      max7456_write(MAX7456_REG_CMAL, i); // set start address low
+      max7456_write(MAX7456_REG_CMDI, characterBitmap[i]);
+    }
+
+    // transfer character buffer from shadow ram to NVM
+    max7456_write(MAX7456_REG_CMM, 0xA0); // must use b1010xxxx
+
+    // wait until bit 5 in the status register returns to 0 (12ms)
+    while ((max7456_read(MAX7456_REG_STAT_READ) & MAX7456_STAT_BIT_CHAR_MEM_BUSY) != 0x00);
+
+    delay(20);
+
+    max7456_write(MAX7456_REG_VM0, MAX7456_VM0_BIT_OSD_ENABLE | MAX7456_VM0_BIT_VSYNC_ENABLE | max7456_videoModeMask);
+
+}
+
+void max7456_resetFont(void)
+{
+    LED0_ON;
+
+    const uint8_t *characterBitmap;
+
+    for(int index = 0; index < 256; index++){
+        characterBitmap = &font_max7456_12x18[index * FONT_MAX7456_12x18_BYTES_PER_CHARACTER];
```

Comment:
yeah, but then it has to be included with the distribution and docs written on how to run it, what you need to install on cygwin, etc.  i'd rather spend more time on other areas TBH.

Marking as wontfix for now. :smile: